mkShapesRDF/shapeAnalysis/runner.py

```diff
@@ -866,19 +991,110 @@ def convertResults(self):
+                                            _from = _h.GetBin(i, 0)
+                                            _to = _h.GetBin(i, 1)
+                                        _fold(_h, _to, _from)
+
+                                if fold == 2 or fold == 3:
+                                    for i in range(0, ny + 2):
+                                        _from = 0
+                                        _to = 0
+                                        if i == 0:
+                                            _from = _h.GetBin(nx + 1, i)
+                                            _to = _h.GetBin(nx, i + 1)
+                                        elif i == ny + 1:
+                                            _from = _h.GetBin(nx + 1, i)
+                                            _to = _h.GetBin(nx, i - 1)
+                                        else:
+                                            _from = _h.GetBin(nx + 1, i)
+                                            _to = _h.GetBin(nx, i)
+                                        _fold(_h, _to, _from)
+
+                                    for i in range(0, nx + 2):
+                                        _from = 0
+                                        _to = 0
+                                        if i == 0:
+                                            _from = _h.GetBin(i, nx + 1)
+                                            _to = _h.GetBin(i + 1, nx)
+                                        elif i == ny + 1:
+                                            _from = _h.GetBin(i, nx + 1)
+                                            _to = _h.GetBin(i - 1, nx)
+                                        else:
+                                            _from = _h.GetBin(i, nx + 1)
+                                            _to = _h.GetBin(i, nx)
+                                        _fold(_h, _to, _from)
+
+                                # unroll 2d
+                                _name = _h.GetName()
+                                _h.SetName(_name + "_old")
+                                new_name = (
+                                    _name
+                                    + "_"
+                                    + _h_name
+                                    + "_"
+                                    + sampleName
+                                    + "_"
+                                    + str(index)
                                 )
-                                _h.SetBinContent(0, 0)
-                            if fold == 2 or fold == 3:
-                                lastBin = _h.GetNbinsX()
-                                _h.SetBinContent(
-                                    lastBin,
-                                    _h.GetBinContent(lastBin)
-                                    + _h.GetBinContent(lastBin + 1),
+
+                                _h2 = ROOT.TH1D(
+                                    new_name, _h.GetName(), nx * ny, 1, nx * ny
```

Comment:
shouldn't the unrolled TH1D histogram have `*nx * ny` bins from 1 to `*nx * ny + 1` ? I think there's one bin less because of the missing `+ 1` in the last argument, but please correct me if I misunderstood